search-cache-pipeline.yml

```diff
@@ -58,7 +58,7 @@ steps:
 
 - bash: >
     $(Build.SourcesDirectory)/.dotnet/dotnet $(Build.SourcesDirectory)/artifacts/bin/Microsoft.TemplateSearch.TemplateDiscovery/Debug/net6.0/Microsoft.TemplateSearch.TemplateDiscovery.dll
-    --basePath $(System.DefaultWorkingDirectory)/NugetDownloadDirectory --allowPreviewPacks -v --test --latestVersionToTest $(RepoSdkVersion)
+    --basePath $(System.DefaultWorkingDirectory)/NugetDownloadDirectory --allowPreviewPacks -v --test --latestVersionToTest $(RepoSdkVersion) --diff $(DiffMode)
```

Comment:
I suggest using a variable name that is easier to understand for the unexperienced. Such as `UseExistingDataAsCache`.

Reply:
`UseExistingDataAsCache` is not fully correct.
When using diff mode, the list of packages discovered on NuGet will be compared with previous run results, and if package version is not changed, the package won't be downloaded and rescanned. Basically setting it to `false`, will force rescanning all the packages, even when not changed. 

I may suggest `EnableDiffMode` as improvement, however not sure if it is worth it.

Reply:
> When using diff mode, the list of packages discovered on NuGet will be compared with previous run results, and if package version is not changed, the package won't be downloaded and rescanned.

This sounds like using existing data as cache though, doesn't it? We use the data we calculated in the previous run instead of downloading the package. What do you mean when you say "UseExistingDataAsCache is not fully correct."?

Reply:
I guess confusion is `cache`. 
The output of command is also `search cache`. `UseExistingDataAsCache` sounds to me - as ignore new data - use old one.

Reply:
I will leave the exact naming up to you. But I would prefer that we at least make it `EnableDiffMode` so that it is clearly a boolean.